xla/stream_executor/cuda/ptx_compiler_helpers.cc

```diff
@@ -56,4 +56,24 @@ absl::Status CreateErrorFromPTXASLog(std::string_view log,
   return absl::OkStatus();
 }
 
+// Warns if the ptxas version should be upgraded.
+// Only prints the warning upon the first invocation.
+void WarnIfBadPtxasVersion(const CudaComputeCapability& cc,
+                           SemanticVersion ptxas) {
+  static absl::once_flag run_once;
+  absl::call_once(run_once, [&] {
+    // nvbug 4826023: Occurs on Hopper+ in CUDA versions 12.x up to and
+    // including CUDA 12.6.2; the earliest ptxas release that corresponds to
+    // CUDA 12.6.3 is ptxas 12.6.85.
+    if (cc.major >= 9 && ptxas >= SemanticVersion{12, 0, 0} &&
+        ptxas < SemanticVersion{12, 6, 85}) {
+      LOG(ERROR)
+          << "*** WARNING *** Invoking PTXAS with version " << ptxas
+          << ", which corresponds to a CUDA version <=12.6.2. CUDA versions "
+             "12.x up to and including 12.6.2 miscompile certain edge cases "
+             "around clamping.\nPlease upgrade to CUDA 12.6.3 or newer.";
```

Comment:
Can also add a sentence saying that for libnvptxcompiler and libnvjitlink this warning might be wrong because we can't determine whether we are using a buggy or a non-buggy CUDA 12.6 version? Ideally this additional sentence only shows up if we are at CUDA version 12.6